### ISSUE TITLE
Test failure: FinalizeCacheHeadersAsync_AddsDate_IfNoneSpecified

### ISSUE THREAD

**Issue by aspnet-hello** [aspnet/Home#3484]
This test [fails](http://aspnetci/viewLog.html?buildId=543346&buildTypeId=Releases_22xPublic_HighSierraOsxUniverse) occasionally with the following error:
```
Assert.Equal() Failure
                                 ↓ (pos 24)
Expected: ··· 31 Aug 2018 19:04:37 GMT
Actual:   ··· 31 Aug 2018 19:04:38 GMT
                                 ↑ (pos 24)
   at Microsoft.AspNetCore.ResponseCaching.Tests.ResponseCachingMiddlewareTests.FinalizeCacheHeadersAsync_AddsDate_IfNoneSpecified() in /_/test/Microsoft.AspNetCore.ResponseCaching.Tests/ResponseCachingMiddlewareTests.cs:line 651
--- End of stack trace from previous location where exception was thrown ---
```
Other tests within that build may have failed with a similar message, but they are not listed here. Check the link above for more info.

This test failed on 2.2.

CC @Eilon (because the bot doesn't know who else to pick)

This issue was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
Please use this workflow to address this flaky test issue, including checking applicable checkboxes and filling in the applicable "TODO" entries:

* Is this actually a flaky test?
  * No, this is a regular test failure, fix the test/product (TODO: Link to commit/PR)
  * Yes, proceed below...

* Is this test failure caused by product code flakiness? (Either this product, or another product this test depends on.)
  * [ ] File a bug against the product (TODO: Link to other bug)
  * Is it possible to change the test to avoid the flakiness?
    * Yes? Go to the "Change the test!" section.
    * No?
      * [ ] Disable the test (TODO: Link to PR/commit)
      * [ ] Wait for other bug to be resolved
      * [ ] Wait for us to get build that has the fix
      * [ ] Re-enable our test (TODO: Link to PR/commit)
      * [ ] Close this bug

* Is it that the test itself is flaky? This includes external transient problems (e.g. remote server problems, file system race condition, etc.)
  * Is there is a way to change our test to avoid this flakiness?
    * Yes? Change the test!
      * [ ] Change the test to avoid the flakiness, for example by using a different test strategy, or by adding retries w/ timeouts (TODO: Link to PR/commit)
      * [ ] Run the test 100 times locally as a sanity check.
      * [ ] Close this bug
    * No?
      * Is there any logging or extra information that we could add to make this more diagnosable when it happens again?
        * Yes?
            * [ ] Add the logging (TODO: Link to PR/commit)
        * No?
            * [ ] Delete the test because flaky tests are not useful (TODO: Link to PR/commit)

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by Eilon** [comment]
@muratg please assign.

**Comment by muratg** [comment]
@JunTaoLuo could you take a look please? Looks like a timing issue, but not sure why.

**Comment by JunTaoLuo** [comment]
@Eilon @muratg I've fixed this in 2.2 and master. Should I also make a PR to fix this in 2.1 (1 failure in 8263 runs)? It's a test fix only so I presume it's easy to get approved.

**Comment by JunTaoLuo** [comment]
I'm going to close this issue for now since it's been addressed. We can file a different issue for 2.1 if we decide to fix it there as well.
